broadinstitute/single_cell_portal_core
# Add Sentry logging for JS ErrorBoundary, Google storage, Bard (SCP-4096)

This adds Sentry logging for application errors, Google Storage fetch errors and Bard request errors coming from the JS side and sets the `logger:app-frontend` to enable differentiating between these JS errors and backend errors.

To test:
- Pull this branch
- bundle install and yarn install to ensure the react-sentry stuff is fully added
- Open this file `app/javascript/lib/sentry-logging.js` and comment out lines 45-47
- To see Bard errors I find it simplest to open up `metrics.api` and change line 426 to fetch something like `${bardDomain}/api-this-will-not-work/event/`
- Then just load up your local and check out [Sentry](https://sentry.io/organizations/broad-institute/issues/?environment=development&project=1424198&query=&statsPeriod=24h) where you should see an error posted similar to:
![Screen Shot 2022-05-02 at 11 37 50 AM](https://user-images.githubusercontent.com/54322292/166318727-a3e99e38-b4ee-4bbc-95ee-3da909ceaff4.png)

to see a different sort of error I found it easiest to:
- open up `StudyResults.jsx` and change line 77 to something like: `<StudyComponentIdonotExist`
- then load up local host and watch Sentry to see an error come through like:
![Screen Shot 2022-05-02 at 4 16 55 PM](https://user-images.githubusercontent.com/54322292/166319471-34e8a5c1-fded-4665-b9a2-f88171a275fc.png)

The logger tag in Sentry will be set to 'app-frontend'

![Screen Shot 2022-05-02 at 11 37 56 AM](https://user-images.githubusercontent.com/54322292/166319231-a86014c1-e525-416a-9629-7a64699b2656.png)
![Screen Shot 2022-05-02 at 11 38 10 AM](https://user-images.githubusercontent.com/54322292/166319232-93441f48-1eed-442d-bd0e-aaf57f0f9d42.png)
![Screen Shot 2022-05-02 at 11 38 19 AM](https://user-images.githubusercontent.com/54322292/166319234-f8a781c9-40e7-4541-b9ff-2afd083fe035.png)
